pkg/types/jsonrpc/codec.go

```diff
@@ -0,0 +1,58 @@
+package jsonrpc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+type Codec struct {
+}
+
+func (*Codec) DecodeRequest(requestBytes []byte, jwtTokenFromHeader string) (Request, error) {
```

Comment:
I don't fully follow - is this the new code that @DeividasK proposed in a draft PR?
Why aren't you moving more of existing Gateway types?